### PR TITLE
Adding vuid 02803

### PR DESCRIPTION
**Pull request by ziga-lunarg**
All elements of pMemoryBarries and pImageMemoryBarriers in CmdWaitEvents must have the same src and dst queue family

### CODE REVIEW

**Comment by jeremyg-lunarg** [comment]
There's a typo in the commit message: All elements of pMemoryBarries and pImageMemoryBarriers

That should be pBufferMemoryBarriers.

Is a similar change needed for CmdWaitEvents2KHR?

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7824,6 +7824,22 @@ bool CoreChecks::PreCallValidateCmdWaitEvents(VkCommandBuffer commandBuffer, uin
     skip |=
         ValidateBarriers(loc.dot(Field::pDependencyInfo), cb_state, srcStageMask, dstStageMask, memoryBarrierCount, pMemoryBarriers,
                          bufferMemoryBarrierCount, pBufferMemoryBarriers, imageMemoryBarrierCount, pImageMemoryBarriers);
+    for (uint32_t i = 0; i < bufferMemoryBarrierCount; ++i) {
+        if (pBufferMemoryBarriers[i].srcQueueFamilyIndex != pBufferMemoryBarriers[i].dstQueueFamilyIndex) {
+            skip |= LogError(commandBuffer, "VUID-vkCmdWaitEvents-srcQueueFamilyIndex-02803",
+                             "vkCmdWaitEvents(): pBufferMemoryBarriers[%" PRIu32
+                             "] has different srcQueueFamilyIndex and dstQueueFamilyIndex.",
```
Comment (sfricke-samsung) [inline layers/core_validation.cpp:7831]: could you print the indexes in the message please

Comment (jeremyg-lunarg) [reply layers/core_validation.cpp:7831]: Took me a second to realize you meant the queue family indices. Yes that'd be nice.

**Comment by ziga-lunarg** [comment]
> There's a typo in the commit message: All elements of pMemoryBarries and pImageMemoryBarriers
> 
> That should be pBufferMemoryBarriers.
> 
> Is a similar change needed for CmdWaitEvents2KHR?

Doesn't seem like CmdWaitEvents2KHR has a similar VUID.

**Comment by jeremyg-lunarg** [comment]
> Doesn't seem like CmdWaitEvents2KHR has a similar VUID.

Ah yes. That's because CmdSetEvent2KHR allows you to pass in a BufferMemoryBarrier so there's enough information for the driver to due a queue family transfer.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM. You could rebase -i and squash the 2 "layers:" commits if you want...

**Pull request merged**